renkuaqs/__init__.py

```diff
@@ -47,28 +51,59 @@ def _check_renku_version():
                      "You should consider install the suggested version.",)
 
 
-class GetGraphHandler(SimpleHTTPRequestHandler):
+# class RenkuFilesWatcher(LoggingEventHandler):
```

Comment:
so this does not work yet, do you think it will work out later?

Reply:
I could see the logging I was expecting anytime a modification was happening to, for example, the folder where the annotations are written.

Problem is that I could not find a way to trigger the refresh of the html page, so I commented in case I found anything.

Reply:
> I could see the logging I was expecting anytime a modification was happening to, for example, the folder where the annotations are written.
>
> Problem is that I could not find a way to trigger the refresh of the html page, so I commented in case I found anything.

You may need to:
* on every update, regenerate the graph and keep it in memory, also keep the version
* add an endpoint to fetch just the graph version
* have a loop/interval in JS which queries the graph version, and if it has changed, reload the page

Reply:
> 
>     * on every update, regenerate the graph and keep it in memory, also keep the version
> 
>     * add an endpoint to fetch just the graph version
> 
what do you mean with the `graph version`?

Reply:
every time graph is updated, git revision is changed, right? Then, git revision would work.

Reply:
ok, yes good idea. I now return also the git revision, along with the `ttl_content` of the graph, this will be used in a comparison (on the JS-frontend side) to understand whether or not the graph has to be rebuilt

Reply:
https://github.com/oda-hub/renku-aqs/issues/44#issuecomment-1439153853